lib/features/support/logger.rb

```diff
@@ -1,9 +1,20 @@
+# frozen_string_literal: true
+
 require 'logger'
 
-if ENV['VERBOSE'] || ENV['DEBUG']
-  $logger = Logger.new(STDOUT, level: Logger::DEBUG)
-elsif ENV['QUIET']
-  $logger = Logger.new(STDOUT, level: Logger::ERROR)
-else
-  $logger = Logger.new(STDOUT, level: Logger::INFO)
-end
\ No newline at end of file
+# A logger, with level configured according to the environment
+class MazeLogger < Logger
+  def initialize
+    if ENV['VERBOSE'] || ENV['DEBUG']
+      super(STDOUT, level: Logger::DEBUG)
+    elsif ENV['QUIET']
+      super(STDOUT, level: Logger::ERROR)
+    else
+      super(STDOUT, level: Logger::INFO)
+    end
+    self.datetime_format = '%Y-%m-%d %H:%M:%S'
+  end
+end
+
+# TODO Added for backward compatibility, but remove in time
```

Comment:
If this global variable is going to be removed in time, is it worth showing the new way of doing it in the above `diagnostics.rb` addition as a reference for later changes?

Reply:
That's a good idea and to be honest I hadn't yet worked out the right way to do it in Ruby.  A singleton seems reasonable, though, so have now update it.  I would have simply updated the logger across the board if it were not for a unit test that mocks it - I didn't want to stretch the scope of this PR too much.